gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/DataFetcherNode.java

```diff
@@ -29,8 +29,8 @@ public DataFetcherNode(String source, Filter filter) {
   public DataFetcherNode(
       String source,
       Filter filter,
-      int limit,
-      int offset,
+      Integer limit,
```

Comment:
why?

Reply:
I had to make this change, so that we can create a new `DataFetcherNode` in `FilterOptimizingVisitor`

```
return
                          new DataFetcherNode(
                              source,
                              filter,
                              dataFetcherNode.getLimit(),
                              dataFetcherNode.getOffset(),
                              dataFetcherNode.getOrderByExpressionList());
```